global/test/TestCommandLineArgumentsTutorial.hpp

```diff
@@ -0,0 +1,173 @@
+   software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE
+GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+*/
+
+#ifndef TESTCOMMANDLINEARGUMENTSTUTORIAL_HPP_
+#define TESTCOMMANDLINEARGUMENTSTUTORIAL_HPP_
+
+#include <cxxtest/TestSuite.h>
+#include <iostream>
+#include <iomanip>
+/* Most Chaste code uses PETSc to solve linear algebra problems.  This involves starting PETSc at the beginning of a test-suite
+ * and closing it at the end.  (If you never run code in parallel then it is safe to replace PetscSetupAndFinalize.hpp with FakePetscSetup.hpp)
+ */
+#include "PetscSetupAndFinalize.hpp"
+#include "CommandLineArguments.hpp"
+
+/**
+ * @file
+ *
+ * This is a tutorial for providing command line arguements to a test from a bash script.
+ * The example bash script also includes a for loop as an example of how such input could
+ * be used to run multiple tests with varying parameter inputs.
+ */
+
+ // NOTE: These tests will run if executed from terminal directly with their default varaible values. However, these
+ // tutorial tests are intended to be executed with the accompanying runcommandlinetutorial.sh bash script.
+ // Investiagte the accompanying runcommandlinetutorial.sh bash script to see how this tutorial should be executed.
+class TestCommandLineArgumentsTutorial : public CxxTest::TestSuite
+{
+public:
+
+    /* The Default test should be ran with the following bash script starting from line 16.
+    runcommandlinetutorial.sh by default has this script already. However, if you have changed
+    the bash script for another text simply copy and paste between START and END back into the file
+    **START**
+      # Here we will declare some values we wish to later pass to a for loop.  
+      N=2
+      L=3
+      M=4
+
+      # Here we set up a simple for loop over variables i,j and k based on the values of N,L and M.
+      for ((i = 0; i <= N; i += 1)); do
+        for ((j = 1; j <= L; j += 1)); do
+          for ((k = 2; k <= M; k += 1)); do
+          # Each loop runs an instance of the TestCommandLineArgumentsTutorial with opt1,opt2 and opt3 taking on the 
+          # values of i,j and k resepctivley.      
+          ~/build/global/test/TestCommandLineArgumentsTutorial -opt1 $i -opt2 $j -opt3 $k &
```

Comment:
Path here is hardcoded, suggest altering instructions to say it should be run "within the build folder". If so will this work?:
```
./global/test/TestCommandLineArgumentsTutorial <args>
```
Test it first(!)